core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -60,6 +60,9 @@ oppia.factory('AnswerClassificationService', [
           if (rule.type !== RULE_TYPE_CLASSIFIER &&
               interactionRulesService[rule.type](
                 answer, rule.inputs)) {
+            if (answerGroups[i].labelledAsCorrect){
+              footerCorrectnessService.setSubmittedAnswerIsCorrect(true);
```

Comment:
The modelling here seems a bit off to me. This method shouldn't have side-effects. I think it's better to update the definition of the AnswerClassificationResultObjectFactory instead, pass the whole thing back to the conversation skin directive, and have that directive orchestrate the behaviour of elements on that page. Otherwise this could be triggered unwantedly in other calls to this service, such as when the creator is checking whether a solution is correct in the editor view.